src/marqo/tensor_search/on_start_script.py

```diff
@@ -185,6 +191,72 @@ def run(self):
             self.logger.info(message)
         self.logger.info("completed loading models")
 
+class PrewarmPatchModels:
+    """Prewarm patch models"""
+
+    logger = get_logger('PrewarmPatchModels')
+    lock = Lock()
+
+    def __init__(self):
+        warmed_models = utils.read_env_vars_and_defaults(EnvVars.MARQO_PATCH_MODELS_TO_PRELOAD)
+        if warmed_models is None:
+            self.models = []
+        elif isinstance(warmed_models, str):
+            try:
+                self.models = json.loads(warmed_models)
+            except json.JSONDecodeError as e:
+                raise exceptions.EnvVarError(
+                    f"Could not parse environment variable `{EnvVars.MARQO_PATCH_MODELS_TO_PRELOAD}`. "
+                    f"Please ensure that this is a JSON-encoded list of strings."
+                ) from e
+        elif isinstance(warmed_models, list):
+            self.models = warmed_models
+        else:
+            raise exceptions.EnvVarError(
+                f"Environment variable `{EnvVars.MARQO_PATCH_MODELS_TO_PRELOAD}` should be a list of strings."
+            )
+        
+        for model in self.models:
+            if model not in ['simple', 'overlap', 'fastercnn', 'frcnn', 'marqo-yolo', 'yolox', 'dino-v1', 'dino-v2', 'dino/v1', 'dino/v2']:
+                raise exceptions.EnvVarError(
+                    f"Invalid model: {model}. Please ensure that this is a valid model."
```

Comment:
Maybe specify it's an invalid patch model, not just that it's an invalid model.

Reply:
done